Travis CI reports success even if compilation fails

Take the recent Linux build on master for example:
https://travis-ci.org/citra-emu/citra/jobs/290731155

Although there is a fatal error (this is also another issue #3023):
```
/citra/src/citra_qt/main.cpp: In member function 'void GMainWindow::InitializeHotkeys()':

/citra/src/citra_qt/main.cpp:248:54: error: 'Cancel' is not a member of 'QKeySequence'

RegisterHotkey("Main Window", "Exit Fullscreen", QKeySequence::Cancel, Qt::ApplicationShortcut);

^

src/citra_qt/CMakeFiles/citra-qt.dir/build.make:682: recipe for target 'src/citra_qt/CMakeFiles/citra-qt.dir/main.cpp.o' failed

make[2]: *** [src/citra_qt/CMakeFiles/citra-qt.dir/main.cpp.o] Error 1

make[2]: *** Waiting for unfinished jobs....

make[1]: *** [src/citra_qt/CMakeFiles/citra-qt.dir/all] Error 2

CMakeFiles/Makefile2:1200: recipe for target 'src/citra_qt/CMakeFiles/citra-qt.dir/all' failed

make: *** [all] Error 2

Makefile:140: recipe for target 'all' failed
```

It still reports success and shows a green icon anyway
```
Done. Your build exited with 0.
```

<!---

Please read the FAQ:
https://citra-emu.org/wiki/faq/

THIS IS NOT A SUPPORT FORUM, FOR SUPPORT GO TO:
https://community.citra-emu.org/

If the FAQ does not answer your question, please go to:
https://community.citra-emu.org/

====================================================

When submitting an issue, please check the following:

- You have read the above.
- You have provided the version (commit hash) of Citra you are using.
- You have provided sufficient detail for the issue to be reproduced.
- You have provided system specs (if relevant).
- Please also provide:
  - For crashes, a backtrace.
  - For graphical issues, comparison screenshots with real hardware.
  - For emulation inaccuracies, a test-case (if able).

--->

--- a/.travis/linux/build.sh
+++ b/.travis/linux/build.sh
@@ -1,3 +1,3 @@
 #!/bin/bash -ex
 
-docker run -v $(pwd):/citra ubuntu:16.04 /bin/bash /citra/.travis/linux/docker.sh
+docker run -v $(pwd):/citra ubuntu:16.04 /bin/bash -ex /citra/.travis/linux/docker.sh

--- a/src/citra_qt/main.cpp
+++ b/src/citra_qt/main.cpp
@@ -245,7 +245,8 @@ void GMainWindow::InitializeHotkeys() {
     RegisterHotkey("Main Window", "Swap Screens", QKeySequence::NextChild);
     RegisterHotkey("Main Window", "Start Emulation");
     RegisterHotkey("Main Window", "Fullscreen", QKeySequence::FullScreen);
-    RegisterHotkey("Main Window", "Exit Fullscreen", QKeySequence::Cancel, Qt::ApplicationShortcut);
+    RegisterHotkey("Main Window", "Exit Fullscreen", QKeySequence(Qt::Key_Escape),
+                   Qt::ApplicationShortcut);
     LoadHotkeys();
 
     connect(GetHotkey("Main Window", "Load File", this), SIGNAL(activated()), this,

